solr/core/src/java/org/apache/solr/cluster/placement/plugins/MinimizeCoresPlacementFactory.java

```diff
@@ -40,13 +40,23 @@
  *
  * <p>See {@link AffinityPlacementFactory} for a more realistic example and documentation.</p>
  */
-public class MinimizeCoresPlacementFactory implements PlacementPluginFactory {
+public class MinimizeCoresPlacementFactory implements PlacementPluginFactory<PlacementPluginFactory.NoConfig> {
 
   @Override
-  public PlacementPlugin createPluginInstance(PlacementPluginConfig config) {
+  public PlacementPlugin createPluginInstance() {
     return new MinimizeCoresPlacementPlugin();
   }
 
+  @Override
+  public void configure(NoConfig cfg) {
+    // no-op
+  }
+
+  @Override
+  public NoConfig getConfig() {
```

Comment:
Maybe similarly here, `PlacementPluginFactory` should provide a default implementation returning `NoConfig.INSTANCE` given that out of 4 implementations of the interface, only one returns something else (`AffinityPlacementFactory`).

Reply:
I think it needs to return null here, I don't see any other way to return an instance that is assignable without ClassCastException to any `<T extends PlacementPluginConfig>`.